### PR TITLE
Fix for ckeditor always marking a form as dirty

### PR DESCRIPTION
**Pull request by kupuguy**
This solvesissue #42 

angular-ckeditor should not mark the control or angular form as dirty on first load.

The solution used here is to check for the first dataReady event and if the control is still pristine set it temporarily to dirty and then restore the pristine state after setting the model data.

This works because Angular only makes the form dirty when a control changes from pristine to dirty so by manually setting the dirty flag the change in state isn't seen outside the control and may be easily reset.

Subsequent changes to the data don't mess with the flags on the control so will work as expected.

### CODE REVIEW

File: angular-ckeditor.js [hunk 1]
```diff
@@ -41,9 +41,26 @@
...
+          var firstTime = true;
+          ['dataReady'].forEach(function (event) {
             controller.onCKEvent(event, function syncView() {
+              // First time only that we receive the dataReady event and the control is pristine it should remain so
+              firstTime &= ngModelController.$pristine;
+              if (firstTime) {
+                ngModelController.$pristine = false;  // Prevent the model change setting the form dirty.
+              }
               ngModelController.$setViewValue(controller.instance.getData() || '');
+              ngModelController.$commitViewValue();
```
Comment (indvd00m) [inline angular-ckeditor.js:59]: `$commitViewValue` first appeared in 1.3, so this is break compatibility with 1.2.x.

**Comment by jwahr** [comment]
Just wondering when this will be merged? Would like to see this get solved. Thanks

**Comment by jwar-gilson** [comment]
@mariusbrn 
Just wondering if this will be merged? Would like to see this get solved. Thanks :)

**Comment by jwar-gilson** [comment]
@JBustin @kupuguy @mariusbrn 

Bueller? 😆

File: angular-ckeditor.js (hunk 1 above)
Comment (pietruz) [reply angular-ckeditor.js:59]: I need this merge, please!

**Comment by gregberge** [comment]
Hello, sorry this package is no longer maintained. Feel free to fork it.

**Pull request closed without merging**